scanpy_scripts/lib/_filter.py

```diff
@@ -53,20 +54,24 @@ def filter_anndata(
     conditions, qc_vars, pct_top = _get_filter_conditions(
         attributes, param, category, subset)
 
-    if 'n_genes' not in adata.obs.columns:
-        sc.pp.filter_cells(adata, min_genes=0)
-    if 'n_counts' not in adata.obs.columns:
-        sc.pp.filter_cells(adata, min_counts=0)
-    if 'n_cells' not in adata.var.columns:
-        sc.pp.filter_genes(adata, min_cells=0)
-    if 'n_counts' not in adata.var.columns:
-        sc.pp.filter_genes(adata, min_counts=0)
-
-    if qc_vars or pct_top:
-        if not pct_top:
-            pct_top = [50]
```

Comment:
This was necessary when a bug exists in sc.pp.calculate_qc_matrics() in earlier version of scanpy, where it failed when pct_top is an empty list. The bug was fixed before scanpy-1.4.3, thus removing this default value.